dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/WebContainer.java

```diff
@@ -1568,41 +1580,56 @@ protected void unsetCacheServletWrapperFactory(CacheServletWrapperFactory factor
         // no-op intended here to avoid cacheServletWrapperFactory being null when switching service implementations
     }
 
-    
-    @Reference(service=ServletVersion.class, cardinality=ReferenceCardinality.MANDATORY, policy=ReferencePolicy.DYNAMIC, policyOption=ReferencePolicyOption.GREEDY)
-    protected synchronized void setVersion(ServiceReference<ServletVersion> reference) {
+   /*
+    * Should only be called by loadServletSpecVersion
+    */
+    protected static void setVersion(int version){
         String methodName = "setVersion";
-        versionRef = reference;
-        WebContainer.loadedContainerSpecLevel = (Integer) reference.getProperty("version");
+        WebContainer.loadedContainerSpecLevel = version;
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
             Tr.debug(tc, methodName, "loadedContainerSpecLevel [ " + WebContainer.loadedContainerSpecLevel + " ]");
         }
     }
 
-    protected synchronized void unsetVersion(ServiceReference<ServletVersion> reference) {
-        if (reference == this.versionRef) {
-            versionRef = null;
-            WebContainer.loadedContainerSpecLevel = DEFAULT_MAX_VERSION;
-        }
+   /*
+    * Not sure if this should exist or if it should be called when webcontainer is deactivating? 
+    */
+    protected static void unsetVersion(int version){
```

Comment:
We need to investigate this and determine the best path forward.

Reply:
Does this actually get called? To me, it would seem it needs to be explicitly called. Previously it would have been called by OSGI.

Reply:
When the servlet feature is change, a new WebContainer class is loaded.  I'm not sure if it's necessary to call unsetVersion? However, it probably probably wouldn't hurt to call this unsetVersion in the `deactivate` method.  I can test it.